packages/sdk/src/types/internal/context.ts

```diff
@@ -182,6 +164,42 @@ export class InternalContext {
 		return actor;
 	}
 
+	public CreateFromGltf(container: AssetContainer, options: {
+		uri: string,
+		colliderType?: 'box' | 'mesh',
+		actor?: Partial<ActorLike>
+	}): Actor {
+		// reserve actor immediately so the pending actor is ready for commands
+		this.protocol.sendPayload({
+			type: 'x-reserve-actor',
+			actor: options.actor
+		} as Payloads.XReserveActor);
+
+		// create actor locally
+		options.actor = Actor.sanitize({ ...options.actor, id: UUID() });
+		this.updateActors(options.actor);
+		const actor = this.context.actor(options.actor.id);
+
+		// kick off asset loading
+		container.loadGltf(options.uri, options.colliderType)
+		.then(assets => {
+			const prefab = assets.find(a => !!a.prefab);
+			if (!prefab) {
+				actor.internal.notifyCreated(false, `glTF file specifies no scenes: ${options.uri}`);
+				return;
+			}
+
+			// once assets are done, find first prefab and spawn it
+			this.CreateFromPrefab({

```

Comment:
>CreateFromPrefab [](start = 8, length = 16)

Problem: CreateFromPrefab is going to create a new Actor with a different id.

Reply:
Damn, you're right. I think everything works as is, except for the UUID. I'll modify it to only add a new ID if it's missing one.

---
In reply to: [316824146](https://github.com/microsoft/mixed-reality-extension-sdk/pull/423#discussion_r316824146) [](ancestors = 316824146)

Reply:
I don't think this is fixed yet. Maybe make a private method they both call?

---
In reply to: [316851109](https://github.com/microsoft/mixed-reality-extension-sdk/pull/423#discussion_r316851109) [](ancestors = 316851109,316824146)

Reply:
That private method they both call *is* createActorFromPayload. All the individual Create calls do is build the payload. I looked into it, but there is no shared code between CreateFromPrefab and this if it's not generating a UUID.

---
In reply to: [316930810](https://github.com/microsoft/mixed-reality-extension-sdk/pull/423#discussion_r316930810) [](ancestors = 316930810,316851109,316824146)

Reply:
However it's accomplished, we need to make certain that application code cannot specify or override the new actor's id.

---
In reply to: [317196273](https://github.com/microsoft/mixed-reality-extension-sdk/pull/423#discussion_r317196273) [](ancestors = 317196273,316930810,316851109,316824146)